src/Speller.php

```diff
@@ -146,16 +150,19 @@ public static function spellCurrency($amount, $language, $currency, $requireDeci
 		
 		$text = trim($speller->parseInt($wholeAmount, false, $currency))
 			. ' '
-			. $speller->getCurrencyName('whole', $wholeAmount, $currency);
+			. ($currencyAsCode 
+				? $currency 
+				: $speller->getCurrencyName('whole', $wholeAmount, $currency));
 		
 		if ($requireDecimal || ($decimalAmount > 0))
 		{
-			$text .= $speller->decimalSeparator
-				. ($spellDecimal
-					? trim($speller->parseInt($decimalAmount, true, $currency))
-					: $decimalAmount)
-				. ' '
-				. $speller->getCurrencyName('decimal', $decimalAmount, $currency);
+			$text .= ($currencyAsCode 
```

Comment:
This doesn't match the contract of `currencyAsCode`, which should be responsible only for preventing spelling the currency code.
Essentially, this should require a different flag, and since there are now quite a few of them, I've been planning to rework the boolean parameters to bitwise flags, e.g. `Speller::spellCurrency(1.23, Speller::LANGUAGE_EN, Speller::CURRENCY_EUR, Speller::FLAG_REQUIRE_DECIMAL | Speller::FLAG_CURRENCY_AS_CODE)`.
This does, unfortunately, mean that it would be problematic for your Twig extension, I've no idea how to make it translate well there.